plugins/inputs/kube_inventory/kube_state.go

```diff
@@ -0,0 +1,179 @@
+
+var sampleConfig = `
+  ## URL for the kubelet
+  url = "https://127.0.0.1"
+
+  ## Namespace to use
+  # namespace = "default"
+
+  ## Use bearer token for authorization. ('bearer_token' takes priority)
+  # bearer_token = "/path/to/bearer/token"
+  ## OR
+  # bearer_token_string = "abc_123"
+
+  ## Set response_timeout (default 5 seconds)
+  # response_timeout = "5s"
+
+  ## Optional Resources to exclude from gathering
+  ## Leave them with blank with try to gather everything available.
+  ## Values can be - "daemonsets", deployments", "nodes", "persistentvolumes",
+  ## "persistentvolumeclaims", "pods", "statefulsets"
+  # resource_exclude = [ "deployments", "nodes", "statefulsets" ]
+
+  ## Optional Resources to include when gathering
+  ## Overrides resource_exclude if both set.
+  # resource_include = [ "deployments", "nodes", "statefulsets" ]
+
+  ## Optional max age for config map
+  # max_config_map_age = "1h"
+
+  ## Optional TLS Config
+  # tls_ca = "/path/to/cafile"
+  # tls_cert = "/path/to/certfile"
+  # tls_key = "/path/to/keyfile"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+
+// SampleConfig returns a sample config
+func (ki *KubernetesInventory) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the description of this plugin
+func (ki *KubernetesInventory) Description() string {
+	return "Read metrics from the Kubernetes api"
+}
+
+// Gather collects kubernetes metrics from a given URL.
+func (ki *KubernetesInventory) Gather(acc telegraf.Accumulator) (err error) {
+	if ki.client == nil {
+		if ki.client, err = ki.initClient(); err != nil {
+			return err
+		}
+	}
+
+	var wg sync.WaitGroup
+
+	if len(ki.ResourceInclude) == 0 {
+		for _, f := range availableCollectors {
+			ctx := context.Background()
+			wg.Add(1)
+			go func(f func(ctx context.Context, acc telegraf.Accumulator, k *KubernetesInventory)) {
+				defer wg.Done()
+				f(ctx, acc, ki)
+			}(f)
+		}
+	} else {
+		for _, n := range ki.ResourceInclude {
+			ctx := context.Background()
+			wg.Add(1)
+			go func(f func(ctx context.Context, acc telegraf.Accumulator, k *KubernetesInventory)) {
+				defer wg.Done()
+				f(ctx, acc, ki)
+			}(availableCollectors[n])
+		}
+	}
+
+	wg.Wait()
+	ki.firstTimeGather = false
+
+	return nil
+}
+
+var availableCollectors = map[string]func(ctx context.Context, acc telegraf.Accumulator, ki *KubernetesInventory){
+	"daemonsets":             collectDaemonSets,
+	"deployments":            collectDeployments,
+	"nodes":                  collectNodes,
+	"persistentvolumes":      collectPersistentVolumes,
+	"persistentvolumeclaims": collectPersistentVolumeClaims,
+	"pods":                   collectPods,
+	"statefulsets":           collectStatefulSets,
+}
+
+func (ki *KubernetesInventory) initClient() (*client, error) {
+	ki.firstTimeGather = true
+
+	if len(ki.ResourceInclude) == 0 {
+		for i := range ki.ResourceExclude {
+			delete(availableCollectors, ki.ResourceExclude[i])
```

Comment:
This will modify the package level variable and take effect for all plugin instances.